lib/radbeacon/le_scanner.rb

```diff
@@ -7,9 +7,26 @@ def initialize(duration = 5)
       @duration = duration
     end
 
+    def scan_command
+      rout, wout = IO.pipe
+      scan_command_str = "sudo hcitool lescan"
+      pid = Process.spawn(scan_command_str, :out => wout)
+      begin
+        Timeout.timeout(@duration) do
+          Process.wait(pid)
+        end
+      rescue Timeout::Error
+        Process.kill('INT', pid)
```

Comment:
Understand, but I think in this case you don't want to be gentle -- the process exceeded it's time out and should be terminated. If you wanted to add a secondary timeout (which I do not recommend) you might even have that send a `KILL` and force the process to stop immediately.